src/main/java/com/checkmarx/flow/service/SastScanner.java

```diff
@@ -67,22 +76,38 @@ public ScanResults scan(ScanRequest scanRequest) {
             Integer projectId = cxScanParams.getProjectId();
 
             log.info("Checking if there is any existing scan for Project: {}", projectId);
-            Integer existingScanId = cxService.getScanIdOfExistingScanIfExists(projectId);
+                Integer existingScanId = cxService.getScanIdOfExistingScanIfExists(projectId);
+
+            String scanComment = getComment(scanRequest);
+            if(cxProperties.getEnablePostActionMonitor()) {
+                //scanComment += (";" + scanRequest.getTeam() + ";");
+                scanComment += (";" + scanRequest.getNamespace() + ";");
+                scanComment += (scanRequest.getApplication() + ";");
+                scanComment += (scanRequest.getBranch() + ";");
+                scanComment += (scanRequest.getMergeNoteUri() + ";");
+                scanComment += (scanRequest.getAdditionalMetadata("statuses_url") + ";");
+                BugTracker bt = scanRequest.getBugTracker();
+                if(bt.getType().getType().equals("GITHUBPULL")) {
+                    scanComment += "PULL";
+                } else {
+                    scanComment += "PUSH";
+                }
```

Comment:
I would put the code of the comments details in separate method to keep the scan method cleaner